rust/connlib/snownet/src/ringbuffer.rs

```diff
@@ -13,8 +13,9 @@ impl<T: PartialEq> RingBuffer<T> {
     }
 
     pub fn push(&mut self, item: T) {
-        if self.buffer.len() == self.buffer.capacity() {
-            // Remove the oldest element (at the beginning) if at capacity
+        let capacity = self.buffer.capacity();
```

Comment:
How else would you solve it? AFAIK, `capacity` only changes when you push an object that would outgrow the current capacity so it kind of exactly tracks what we want?

Reply:
I'd keep a separate variable, since `capacity` is just the memory it has allocated, which it coincides right now in our implementation but we might want to create an underlying buffer with a smaller capacity and grow it up to a point.

I don't feel very strongly about this, we can leave it like this.